addons/sale_mrp/models/sale_mrp.py

```diff
@@ -60,7 +60,8 @@ def _compute_qty_delivered(self):
                 # when the product sold is made only of kits. In this case, the BOM of the stock moves
                 # do not correspond to the product sold => no relevant BOM.
                 elif boms:
-                    if all([m.state == 'done' for m in order_line.move_ids]):
+                    # if the move is ingoing, the product **sold** has delivered qty 0
+                    if all([m.state == 'done' and m.picking_code == "outgoing" for m in order_line.move_ids]):
```

Comment:
check the m location dest usage instead of picking type. There is no constraint on ongoing to go to customer

Reply:
and when the user will validate the return the quantity delivered will remain the order line quantity. Don't spend too much energy fixing it